src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/Reliability/SqlRetryPolicy.Generic.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+using System;
+
+namespace Microsoft.Data.SqlClient.Reliability
+{
+    /// <summary>
+    /// Provides a generic version of the <see cref="SqlRetryPolicy"/> class.
+    /// </summary>
+    /// <typeparam name="T">The type that implements the <see cref="ITransientErrorDetectionStrategy"/> interface that is responsible for detecting transient conditions.</typeparam>
+    public class SqlRetryPolicy<T> : SqlRetryPolicy where T : ITransientErrorDetectionStrategy, new()
```

Comment:
Why does this `T` have to be `new()`. Is it not enough that callers pass in a value of type `T`? The caller can instantiate that type. This would reduce the dependencies that this class requires (dependency injection).